src/libcollectionstest/str.rs

```diff
@@ -786,9 +786,10 @@ fn test_rev_iterator() {
 
 #[test]
 fn test_chars_decoding() {
+    let mut bytes = [0; 4];
     for c in (0..0x110000).filter_map(::std::char::from_u32) {
-        let bytes = c.encode_utf8();
-        let s = ::std::str::from_utf8(bytes.as_slice()).unwrap();
+        let b = c.encode_utf8(&mut bytes).as_bytes();
+        let s = ::std::str::from_utf8(b).unwrap();
```

Comment:
I think I kept the `from_utf8()` because it also tests `encode_utf8()`, but I'll move it to the right place.